codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/ServerCombinedErrorGeneratorTest.kt

```diff
@@ -0,0 +1,87 @@
+import software.amazon.smithy.rust.codegen.testutil.TestWorkspace
+import software.amazon.smithy.rust.codegen.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.testutil.compileAndTest
+import software.amazon.smithy.rust.codegen.testutil.renderWithModelBuilder
+import software.amazon.smithy.rust.codegen.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.testutil.unitTest
+import software.amazon.smithy.rust.codegen.util.lookup
+
+class ServerCombinedErrorGeneratorTest {
+    private val baseModel = """
+        namespace error
+
+        operation Greeting {
+            errors: [InvalidGreeting, ComplexError, FooException]
+        }
+
+        @error("client")
+        @retryable
+        structure InvalidGreeting {
+            @required
+            message: String,
+        }
+
+        @error("server")
+        structure FooException { }
+
+        @error("server")
+        structure ComplexError {
+            abc: String,
+            other: Integer
+        }
+    """.asSmithyModel()
+    private val model = OperationNormalizer.transform(baseModel)
+    private val symbolProvider = serverTestSymbolProvider(model)
+
+    @Test
+    fun `generates combined error enums`() {
+        val project = TestWorkspace.testProject(symbolProvider)
+        project.withModule(RustModule.public("error")) { writer ->
+            listOf("FooException", "ComplexError", "InvalidGreeting").forEach {
+                model.lookup<StructureShape>("error#$it").renderWithModelBuilder(model, symbolProvider, writer, TYPE.SERVER)
+            }
+            val generator = ServerCombinedErrorGenerator(model, symbolProvider, model.lookup("error#Greeting"))
+            generator.render(writer)
+
+            writer.unitTest(
+                name = "generates_combined_error_enums",
+                test = """
+                    let variant = InvalidGreeting::builder().message("an error").build();
+                    assert_eq!(format!("{}", variant), "InvalidGreeting: an error");
+                    assert_eq!(variant.message(), "an error");
+                    assert_eq!(
+                        variant.retryable_error_kind(),
+                        aws_smithy_types::retry::ErrorKind::ClientError
+                    );
+
+                    let error = GreetingError::InvalidGreeting(variant);
+
+                    // Generate is_xyz methods for errors.
+                    assert_eq!(error.is_invalid_greeting(), true);
+                    assert_eq!(error.is_complex_error(), false);
+
+                    // Indicate the original name in the display output.
+                    let error = FooException::builder().build();
+                    assert_eq!(format!("{}", error), "FooException")
+                """
+            )
+
+            println("file:///${project.baseDir}/src/lib.rs")
+            println("file:///${project.baseDir}/src/error.rs")
```

Comment:
Why this print? Is it a leftover or is it there for debugging purposes?

Reply:
@rcoh Is there a way of having the test infrastructure be smarter about these and always print all the generated sources automatically?

Reply:
I've cut https://github.com/awslabs/smithy-rs/issues/1178 to track this improvement.